### PR TITLE
chore(docs): Correct image guide on CSS-in-JS usage

### PR DESCRIPTION
**Pull request by LekoArts**
## Description

Our current documentation is slightly incorrect. It was probably assumed that the `css` prop behaves the same across libraries, however that's not the case.

While Emotion generates a `React.createElement` call (https://emotion.sh/docs/css-prop#get-started), styled-components just calls `styled` function under the hood (https://styled-components.com/docs/api#css-prop) -- and that styled function is not supported.

I've also added a note about the className props in regards to libraries like e.g. linaria

## Related Issues

Fixes https://github.com/gatsbyjs/gatsby/issues/30880

### CODE REVIEW

File: docs/docs/reference/built-in-components/gatsby-plugin-image.md [hunk 1]
```diff
@@ -110,7 +110,7 @@ export function Dino() {
 }
 ```
 
-You should instead use the `css` prop provided by these libraries:
+If you use Emotion you can use the provided `css` prop:
```
Comment (meganesu) [inline docs/docs/reference/built-in-components/gatsby-plugin-image.md:113]: Trying to connect this paragraph back to the previous one

```suggestion
If you use Emotion you can use the provided `css` prop instead:
```

File: docs/docs/reference/built-in-components/gatsby-plugin-image.md [hunk 2]
```diff
@@ -127,22 +127,9 @@ export function Dino() {
...
-    <StaticImage
-      src="trex.png"
-      css={`
-        border: 4px green dashed;
-      `}
-    />
-  )
-}
-```
+Unfortunately the [`css` prop from styled-components](https://styled-components.com/docs/api#css-prop) turns the code into a `styled` function under the hood and as explained above `StaticImage` doesn't support that syntax.
```
Comment (meganesu) [inline docs/docs/reference/built-in-components/gatsby-plugin-image.md:130]: It feels like there's a second sentence missing from this paragraph. Something about what the implication is for the user. Like, "If you're using styled-components, you should do X instead." Or can you just not use styled-components with it?

File: docs/docs/reference/built-in-components/gatsby-plugin-image.md [hunk 3]
```diff
@@ -127,22 +127,9 @@ export function Dino() {
...
 
-You can also use a regular `style` or `className` prop. Note that in all of these cases the styling is applied to the wrapper, not the image itself. If you need to style the `<img>` tag, you can use `imgStyle` or `imgClassName`.
+You can also use a regular `style` or `className` prop. Note that in all of these cases the styling is applied to the wrapper, not the image itself. If you need to style the `<img>` tag, you can use `imgStyle` or `imgClassName`. The `className` or `imgClassName` prop is helpful if your styling library is giving you a computed class name string instead of the computed styles.
```
Comment (meganesu) [inline docs/docs/reference/built-in-components/gatsby-plugin-image.md:132]: Might be nice to have an example showing what this looks like? Just an idea.

**Review by meganesu — CHANGES_REQUESTED** [review]
Left a few comments. Thanks for clarifying this doc!

File: docs/docs/reference/built-in-components/gatsby-plugin-image.md (hunk 2 above)
Comment (LekoArts) [reply docs/docs/reference/built-in-components/gatsby-plugin-image.md:130]: Yeah, you can't use styled-components with this. That's pretty much the implication.

File: docs/docs/reference/built-in-components/gatsby-plugin-image.md (hunk 3 above)
Comment (LekoArts) [reply docs/docs/reference/built-in-components/gatsby-plugin-image.md:132]: Added an example

**Review by meganesu — APPROVED** [review]

**Pull request merged**